### ISSUE TITLE
Package name containing reserved Java keyword is not cleaned up

### ISSUE THREAD

**Issue by samair** [spring-io/initializr#983]
For java projects, reserved keywords such as *final*, *public* etc are accepted for the artifactId value. When the generated project is imported, it would fail to build.

**Comment by bono007** [comment]
Just so I am clear, its ok for the `artifactId` to have the reserved keywords, but not for the package name. Correct?

I would like to grab this issue @snicoll .

**Comment by bono007** [comment]
### Background
**TLDR;** Any segment of the package name must not be a [Keyword](https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.9) or BooleanLiteral or NullLiteral (`true|false|null`). 

The rules of package names are defined by the [JLS](https://docs.oracle.com/javase/specs/jls/se8/html/jls-7.html#jls-7.4.1) and package names must be made of [valid identifiers](https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.8).

### Plan
If one of the above is encountered, the `io.spring.initializr.metadata.InitializrConfiguration#cleanPackageName(java.lang.String, java.lang.String)` will use the `defaultPackageName` instead (like it does currently w/ the other cleaning).

### Question(s)
1. Should we not log something about why the default package name was used? It looks like it currently just silently "cleans/swaps" it for the other situations. Any reason we don't currently log these "cleans"?

**Comment by bono007** [comment]
This one is ready for review @snicoll

**Comment by snicoll** [comment]
Closing in favour of PR #1018 - thanks @bono007!
